### PR TITLE
[SPARK-20995][CORE]'Spark-env.sh.template' should add 'YARN_CONF_DIR' configuration instructions.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## What changes were proposed in this pull request?

Ensure that `HADOOP_CONF_DIR` or `YARN_CONF_DIR` points to the directory which contains the (client side) configuration files for the Hadoop cluster.
These configs are used to write to HDFS and connect to the YARN ResourceManager. The
configuration contained in this directory will be distributed to the YARN cluster so that all
containers used by the application use the same configuration. 

Sometimes, `HADOOP_CONF_DIR` is set to the hdfs configuration file path. So, YARN_CONF_DIR should be set to the yarn configuration file path.

My project configuration item of 'spark-env.sh ' is as follows:

'HADOOP_CONF_DIR' configuration file path. List the relevant documents below:

'YARN_CONF_DIR' configuration file path. List the relevant documents below:

So, 'Spark-env.sh.template' should add 'YARN_CONF_DIR' configuration instructions.

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: conf/spark-env.sh.template [hunk 1]
```diff
@@ -23,6 +23,7 @@
 # Options read when launching programs locally with
 # ./bin/run-example or ./bin/spark-submit
 # - HADOOP_CONF_DIR, to point Spark towards Hadoop configuration files
+# - YARN_CONF_DIR, to point Spark towards Yarn configuration files
```
Comment (srowen) [inline conf/spark-env.sh.template]: Yarn -> YARN

**Review by srowen — APPROVED** [review]
I think that's fine, because the var is still used. You shouldn't really open a JIRA for this

File: conf/spark-env.sh.template (hunk 1 above)
Comment (guoxiaolongzte) [inline conf/spark-env.sh.template]: Thanks. I will fix it.

File: conf/spark-env.sh.template [hunk 2]
```diff
@@ -23,6 +23,7 @@
 # Options read when launching programs locally with
 # ./bin/run-example or ./bin/spark-submit
 # - HADOOP_CONF_DIR, to point Spark towards Hadoop configuration files
+# - YARN_CONF_DIR, to point Spark towards YARN configuration files
```
Comment (srowen) [inline conf/spark-env.sh.template]: Hang on, I don't think this env variable would be set when not using YARN right? Hadoop conf dirs might, but not YARN.

Comment (guoxiaolongzte) [inline conf/spark-env.sh.template]: When you use ‘spark-submit --master yarn /home/...jar’  to submit the spark job, you need to set the YARN_CONF_DIR value.

Can i modify it like this? as follows:
+# - YARN_CONF_DIR, to point Spark towards YARN configuration files **when you use YARN.**

Comment (guoxiaolongzte) [inline conf/spark-env.sh.template]: ping @srowen

Comment (srowen) [inline conf/spark-env.sh.template]: Sure, but you have added this to the section that doesn't concern YARN. Just remove this line. You added it to the YARN section below.

Comment (guoxiaolongzte) [inline conf/spark-env.sh.template]: I have fixed it according to your suggestion.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**